saphir/src/utils.rs

```diff
@@ -19,29 +25,35 @@ pub struct EndpointResolver {
     path_matcher: UriPathMatcher,
     methods: HashSet<Method>,
     id: u64,
+    allow_any_method: bool,
 }
 
 impl EndpointResolver {
     pub fn new(path_str: &str, method: Method) -> Result<EndpointResolver, SaphirError> {
         let mut methods = HashSet::new();
-        methods.insert(Method::OPTIONS);
+        let allow_any_method = method.is_any();
         methods.insert(method);
 
+
         Ok(EndpointResolver {
             path_matcher: UriPathMatcher::new(path_str).map_err(|e| SaphirError::Other(e))?,
             methods,
-            id: ENDPOINT_ID.fetch_add(1, std::sync::atomic::Ordering::SeqCst)
+            id: ENDPOINT_ID.fetch_add(1, std::sync::atomic::Ordering::SeqCst),
+            allow_any_method
         })
     }
 
     pub fn add_method(&mut self, m: Method) {
+        if !self.allow_any_method && m.is_any() {
+            self.allow_any_method = true;
+        }
         self.methods.insert(m);
```

Comment:
Should put this in an else since adding any to the method insert is taking up memory uselessly (albeit small)

Reply:
Fixed